packages/@aws-cdk/aws-location-alpha/lib/tracker.ts

```diff
@@ -0,0 +1,267 @@
+   * Use an existing tracker by name
+   */
+  public static fromTrackerName(scope: Construct, id: string, trackerName: string): ITracker {
+    const trackerArn = Stack.of(scope).formatArn({
+      service: 'geo',
+      resource: 'tracker',
+      resourceName: trackerName,
+    });
+
+    return Tracker.fromTrackerArn(scope, id, trackerArn);
+  }
+
+  /**
+   * Use an existing tracker by ARN
+   */
+  public static fromTrackerArn(scope: Construct, id: string, trackerArn: string): ITracker {
+    const parsedArn = Stack.of(scope).splitArn(trackerArn, ArnFormat.SLASH_RESOURCE_NAME);
+
+    if (!parsedArn.resourceName) {
+      throw new Error(`Tracker Arn ${trackerArn} does not have a resource name.`);
+    }
+
+    class Import extends Resource implements ITracker {
+      public readonly trackerName = parsedArn.resourceName!;
+      public readonly trackerArn = trackerArn;
+    }
+
+    return new Import(scope, id, {
+      account: parsedArn.account,
+      region: parsedArn.region,
+    });
+  }
+
+  public readonly trackerName: string;
+
+  public readonly trackerArn: string;
+
+  /**
+   * The timestamp for when the tracker resource was created in ISO 8601 format
+   *
+   * @attribute
+   */
+  public readonly trackerCreateTime: string;
+
+  /**
+   * The timestamp for when the tracker resource was last updated in ISO 8601 format
+   *
+   * @attribute
+   */
+  public readonly trackerUpdateTime: string;
+
+  constructor(scope: Construct, id: string, props: TrackerProps = {}) {
+
+    if (props.description && !Token.isUnresolved(props.description) && props.description.length > 1000) {
+      throw new Error(`\`description\` must be between 0 and 1000 characters. Received: ${props.description.length} characters`);
+    }
+
+    if (props.trackerName && !Token.isUnresolved(props.trackerName) && !/^[-.\w]{1,100}$/.test(props.trackerName)) {
+      throw new Error(`Invalid tracker name. The tracker name must be between 1 and 100 characters and contain only alphanumeric characters, hyphens, periods and underscores. Received: ${props.trackerName}`);
+    }
+
+    if (!Token.isUnresolved(props.kmsKey)
+      && !props.kmsKey
+      && props.kmsKeyEnableGeospatialQueries
+    ) {
+      throw new Error('`kmsKeyEnableGeospatialQueries` can only be enabled that are configured to use an AWS KMS customer managed key');
+    }
+
+    super(scope, id, {
+      physicalName: props.trackerName ?? Lazy.string({ produce: () => generateUniqueId(this) }),
+    });
+
+    const tracker = new CfnTracker(this, 'Resource', {
+      trackerName: this.physicalName,
+      description: props.description,
+      eventBridgeEnabled: props.eventBridgeEnabled,
+      kmsKeyEnableGeospatialQueries: props.kmsKeyEnableGeospatialQueries,
+      kmsKeyId: props.kmsKey?.keyArn,
+      positionFiltering: props.positionFiltering,
+    });
+
+    props.geofenceCollections?.forEach((collection) => {
+      new CfnTrackerConsumer(this, `TrackerConsumer${collection.node.id}`, {
```

Comment:
Initially, I had TrackerConsumer as a separate Construct. 
https://github.com/aws/aws-cdk/pull/31268/commits/5fb3cdd5c13b6efda280802bfb316275c407caa3#diff-fc1da6063a6db97ebec475178eaaa40ce1a7d0a56f8c1b5cb249b99f33e9946a

However, since it only involved associations, I decided to integrate it into the Tracker. 
If you have any opinions on this design, please let me know.

Reply:
> However, since it only involved associations, I decided to integrate it into the Tracker.
If you have any opinions on this design, please let me know.

@mazyu36 what do you mean by it only involved associations? Just looking to clarify my understanding. Here it seems CDK users would not be able to create TrackerConsumer objects via an L2 - does this make sense?

Reply:
@sumupitchayan 
TrackerConsumer is just an association between Tracker and Consumer (GeofenceCollection) and has no actual entity.

I thought the association could be auto-generated, similar to how VPC Subnet and route table associations work.
https://github.com/aws/aws-cdk/blob/main/packages/aws-cdk-lib/aws-ec2/lib/vpc.ts#L2117

Alternatively, TrackerConsumer could be made into a separate L2 construct.

Which approach would be better do you think?

Reply:
@mazyu36 thanks for the clarification, this design makes sense to me. From my understanding, customers won't directly be creating TrackerConsumer objects and this case is similar to how VPC Subnets/route table associations are designed.

Additionally, since this is an alpha module and it will be easy to add an L2 for TrackerConsumer for this design, I think we can just do that in the future if we find it is needed for a customer use case.